javascript/javascript3/week1/cityWeather.js

```diff
@@ -0,0 +1,155 @@
+    .then((response)=> response.json())
+    
+    .then((weather)=> {
+    
+        writeDataFromFetch(weather)
+    
+    })
+    .catch((error) => {
+      console.error('This is not a city!!! ', error);
+    })}
+//  fetch weathe by location
+function fetchFromLocation(Latitude,longitude){
+    fetch(`http://api.openweathermap.org/data/2.5/weather?lat=${Latitude}&lon=${longitude}&appid=1fa178d7ac2b383598fae81ce7953bc7&units=metric`)
+    .then((response)=> response.json())
+    
+    .then((weather)=> {
+    
+        writeDataFromFetch(weather);
+    })
+}
+
+ function writeDataFromFetch(weather){
+    newCity.value = weather.name;
+    div.innerHTML = `City : ${weather.name}`+"<br>"
+    div.innerHTML += `Temperature :  ${weather.main.temp}&#176c`+"<br>"
+    const weatherDesc = weather.weather[0].description
+    div.innerHTML += weatherDesc.charAt(0).toUpperCase()+weatherDesc.slice(1) + iconWriter(weatherDesc)+'<br>'
+    div.innerHTML += `Wind speed : ${weather.wind.speed}m/s`+"<br>"
+    div.innerHTML += `It is ${weather.clouds.all}% Coudy`+"<br>"
+    div.innerHTML += `Sunrise at : ${convertToTime(weather.sys.sunrise)}`+"<br>"
+    div.innerHTML += `Sunset at : ${convertToTime(weather.sys.sunset)}`+"<br>"
+
+ }
+
+//  retrieve user loction
+
+function weatherLocationCapture(){
+    if( navigator.geolocation){
+        navigator.geolocation.getCurrentPosition(showThePosition)
+    }
+    else{
+        console.log("Geolocation doesn't support by this browser"); 
+    }  
+}
+
+//    callup fetch with positions
+function showThePosition(position){
+    fetchFromLocation(position.coords.latitude,position.coords.longitude)
+}
+
+
+function saveToLocalhost(){
+// save to local host   newCity.value
+localStorage.setItem("mySavedCity",newCity.value);
+console.log(newCity.value,' Has been saved');
+
+}
+ //          show proper Icon
+function iconWriter(kindOfWeather){
+switch(true){
+    case kindOfWeather.includes("cloud") : return ' <i class="fas fa-cloud"></i> '
+                                            
+    case kindOfWeather.includes("clear sky") : return ' <i class="fas fa-sun"></i> '
+                                            
+    case kindOfWeather.includes("rain") : return ' <i class="fas fa-cloud-showers-heavy"></i> '
+                                              
+    case kindOfWeather.includes("thunderstorm") : return ' <i class="fas fa-poo-storm"></i> '
+
+    case kindOfWeather.includes("snow") : return ' <i class="fas fa-snowflake"></i> '
+                                                
+    case kindOfWeather.includes("mist") : return ' <i class="fas fa-smog"></i> '
+
+    case kindOfWeather.includes("smoke") : return ' <i class="fas fa-smog"></i> '
+
+    case kindOfWeather.includes("wind") : return ' <i class="fas fa-wind"></i> '
+
+    default:        return ' <i class="fas fa-temperature-low"></i> '                                  
+}
+}
```

Comment:
No required to create your own function to render icon, icon is already returned by api. you can create image tag in render function and using "http://openweathermap.org/img/w/${iconCode}.png"  url for src. here "iconCode" is just placeholder, you need to get icon property returned from api and replace in url.